### ISSUE TITLE
hail applies disk policies to total disk size, instead of individual disks

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#474]
Originally reported of Google Code with ID 418.

```
<b>What steps will reproduce the problem?</b>

# gnt-cluster modify --specs-disk-size=min=1024,max=2048
# gnt-instance add -t file --disk 0:size=1536 --disk 1:size=1536 -o busybox aehlig-gnt1


<b>What is the expected output? What do you see instead?</b>

- expected: hail chooses a node and the instance gets created

- found:
Failure: prerequisites not met for this operation:
error type: insufficient_resources, error details:
Can't compute nodes using iallocator 'hail': Request failed: Group default (preferred): No valid allocation solutions, failure reasons: FailDisk: 3

Additional information

- manually choosing a node works, i.e., 

# gnt-instance add -t file --disk 0:size=1536 --disk 1:size=1536 -o busybox -n node1 aehlig-gnt1

  creates the instance

- for too small disk sizes, you have the opposite problem: hail finds a node, and allocation there is violating policy

# gnt-instance add -t file --disk 0:size=768 --disk 1:size=768 -o busybox aehlig-gnt2
Tue Apr  9 09:06:03 2013  - INFO: Selected nodes for instance aehlig-gnt2 via iallocator hail: node3
Failure: prerequisites not met for this operation:
error type: wrong_input, error details:
Instance allocation to group XXX (default) violates policy: disk-size/0 value 768 is not in range [1024, 2048], disk-size/1 value 768 is not in range [1024, 2048]

```
Originally added on 2013-04-09 09:17:50 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-04-10 18:15:28 +0000 UTC.

Added Labels: Priority-Medium
Added to Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
fixed in 241cea1e8d53b7e2f68
```
Originally added on 2013-04-11 17:41:36 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Is there a recommended workaround for this problem in a production cluster?
```
Originally added on 2013-04-17 12:01:03 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I'm not aware of any good work around for multi-disk instances. But, at least,

- it is working for single-disk instances, and
- there is no issue if the question whether an instance can be placed
  on a certain node group is independent on whether individual disks
  or total disk space is taken; an example would be the same, sufficiently
  liberal, policy on every node group.

For the remaining cases, the best I can recommend is to manually place
the instances using the -n option.

Also note that the hail binary can be replaced without updating the rest
of ganeti.
```
Originally added on 2013-04-17 12:24:14 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:49 +0000 UTC.

Changed State: Released
